react/components/SearchQuery.js

```diff
@@ -199,31 +177,11 @@ const useQueries = (
   const productSearchResult = useQuery(productSearchQuery, {
     variables: {
       ...variables,
-      showSponsored:
-        settings?.fetchSponsoredProductsOnSearch &&
-        sponsoredProductsBehavior === 'sync',
+      showSponsored: true,
```

Comment:
if this flag will always be `true` and we check the store in the resolver, can't we just remove the flag here?

Reply:
@thalytafabrine Not yet, because not all placements support ads yet, like shelves and autocomplete. But both of them call the `productSearch` query.